bind/gen.go

```diff
@@ -0,0 +1,666 @@
+	_%[1]s.GoPyInit()
+
+	`
+
+	// 3 = gencmd, 4 = vm, 5 = libext
+	MakefileTemplate = `# Makefile for python interface for package %[1]s.
+# File is generated by gopy. Do not edit.
+# %[2]s
+
+GOCMD=go
+GOBUILD=$(GOCMD) build
+GOIMPORTS=goimports
+PYTHON=%[4]s
+PYTHON_CFG=$(PYTHON)-config
+GCC=gcc
+LIBEXT=%[5]s
+
+# get the flags used to build python:
+CFLAGS = $(shell $(PYTHON_CFG) --cflags)
+LDFLAGS = $(shell $(PYTHON_CFG) --ldflags)
+
+all: gen build
+
+gen:
+	%[3]s
+
+build:
+	# build target builds the generated files -- this is what gopy build does..
+	# this will otherwise be built during go build and may be out of date
+	- rm %[1]s.c
+	# goimports is needed to ensure that the imports list is valid
+	$(GOIMPORTS) -w %[1]s.go
+	# generate %[1]s_go$(LIBEXT) from %[1]s.go -- the cgo wrappers to go functions
+	$(GOBUILD) -buildmode=c-shared -o %[1]s_go$(LIBEXT) %[1]s.go
+	# use pybindgen to build the %[1]s.c file which are the CPython wrappers to cgo wrappers..
+	# note: pip install pybindgen to get pybindgen if this fails
+	$(PYTHON) build.py
+	# build the _%[1]s$(LIBEXT) library that contains the cgo and CPython wrappers
+	# generated %[1]s.py python wrapper imports this c-code package
+	$(GCC) %[1]s.c -dynamiclib %[1]s_go$(LIBEXT) -o _%[1]s$(LIBEXT) $(CFLAGS) $(LDFLAGS) -w
+	
+`
+
+	// exe version of template: 3 = gencmd, 4 = vm, 5 = libext
+	MakefileExeTemplate = `# Makefile for python interface for standalone executable package %[1]s.
+# File is generated by gopy. Do not edit.
+# %[2]s
+
+GOCMD=go
+GOBUILD=$(GOCMD) build
+GOIMPORTS=goimports
+PYTHON=%[4]s
+PYTHON_CFG=$(PYTHON)-config
+GCC=gcc
+LIBEXT=%[5]s
+
+# get the flags used to build python:
+CFLAGS = $(shell $(PYTHON_CFG) --cflags)
+LDFLAGS = $(shell $(PYTHON_CFG) --ldflags)
+
+all: gen build
+
+gen:
+	%[3]s
+
+build:
+	# build target builds the generated files into exe -- this is what gopy build does..
+	# goimports is needed to ensure that the imports list is valid
+	$(GOIMPORTS) -w %[1]s.go
+	# this will otherwise be built during go build and may be out of date
+	- rm %[1]s.c 
+	echo "typedef uint8_t bool;" > %[1]s_go.h
+	# this will fail but is needed to generate the .c file that then allows go build to work
+	- $(PYTHON) build.py >/dev/null 2>&1
+	# generate %[1]s_go.h from %[1]s.go -- unfortunately no way to build .h only
+	$(GOBUILD) -buildmode=c-shared -o %[1]s_go$(LIBEXT) >/dev/null 2>&1
+	# use pybindgen to build the %[1]s.c file which are the CPython wrappers to cgo wrappers..
+	# note: pip install pybindgen to get pybindgen if this fails
+	$(PYTHON) build.py
+	# build the executable
+	- rm %[1]s_go$(LIBEXT)
+	$(GOBUILD) -o py%[1]s
+	
+`
+)
+
+// actually, the weird thing comes from adding symbols to the output, even with dylib!
+// ifeq ($(LIBEXT), ".dylib")
+// 	# python apparently doesn't recognize .dylib on mac, but .so causes extra weird dylib dir..
+// 	- ln -s _%[1]s$(LIBEXT) _%[1]s.so
+// endif
+//
+
+var thePyGen *pyGen
```

Comment:
why do we need this global variable?

Reply:
```Go
// thePyGen is the current pyGen which is needed in symbols to lookup
// package paths -- not very clean to pass around or add to various
// data structures to make local, but if that ends up being critical
// for some reason, it could be done.
var thePyGen *pyGen
```